integration/indexes_compat_test.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+func TestIndexesList(t *testing.T) {
+	t.Helper()
+
+	s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+		Providers:                shareddata.AllProviders(),
+		AddNonExistentCollection: true,
+	})
+	ctx, targetCollections, compatCollections := s.Ctx, s.TargetCollections, s.CompatCollections
+
+	for i := range targetCollections {
+		targetCollection := targetCollections[i]
+		compatCollection := compatCollections[i]
+
+		t.Run(targetCollection.Name(), func(t *testing.T) {
+			t.Helper()
+			t.Parallel()
+
+			targetCur, targetErr := targetCollection.Indexes().List(ctx)
+			compatCur, compatErr := compatCollection.Indexes().List(ctx)
+
+			require.NoError(t, compatErr)
+			assert.Equal(t, compatErr, targetErr)
+
+			targetRes := FetchAll(t, ctx, targetCur)
+			compatRes := FetchAll(t, ctx, compatCur)
+
+			assert.Equal(t, compatRes, targetRes)
+		})
+	}
+}
+
+func TestIndexesCreate(t *testing.T) {
+	t.Helper()
+
+	setup.SkipForTigrisWithReason(t, "Indexes creation is not supported for Tigris")
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		models      []mongo.IndexModel
+		altErrorMsg string                   // optional, alternative error message in case of error
+		resultType  compatTestCaseResultType // defaults to nonEmptyResult
+		skip        string                   // optional, skip test with a specified reason
+	}{
+		"empty": {
+			models:     []mongo.IndexModel{},
+			resultType: emptyResult,
+		},
+		"single-index": {
+			models: []mongo.IndexModel{
+				{Keys: bson.D{{"v", -1}}},
+			},
+		},
+		"duplicate_id": {
+			models: []mongo.IndexModel{
+				{
+					Keys: bson.D{{"_id", 1}}, // this index is already created by default
```

Comment:
I feel index crated by default was unique index, which is the same as ascending index?